utils.py

```diff
@@ -721,6 +722,86 @@ def get_hashable_value(value):
         return value
 
 
+def convert_to_text_angular(html_data):
+    """This fucntion converts the html to text angular supported format.
+
+    Args:
+        html_data: str. HTML string to be converted.
+
+    Returns:
+        str. The converted HTML string.
+    """
+    soup = bs4.BeautifulSoup(html_data, 'html.parser')
+
+    tags_to_remove = ['div', 'span', 'code', 'table', 'td', 'tr', 'tbody']
+
+    for tag_name in tags_to_remove:
+        for tag in soup.findAll(tag_name):
+            tag.unwrap()
+
+    for br in soup.findAll('br'):
+        if br.parent.name not in ['b', 'i', 'li', 'p']:
+            br.wrap(soup.new_tag('p'))
+
+    tag_list = soup.findAll()
+    if not len(tag_list):
+        start_tag = None
+    else:
+        start_tag = '<' + soup.findAll()[0].name + '>'
+    if not start_tag or html_data.find(start_tag) != 0:
```

Comment:
Don't quite understand, what does this block do?

Reply:
If start tag is missing, then the html string is wrapped in p tags.